### ISSUE TITLE
pip install fails when in virtualenv

### ISSUE THREAD

**Issue by tklicki** [Juniper/jsnapy#205]
It seems the problem is with hardcoded '/etc/jsnapy directory'. After creating virtualenv you can't install jsnapy correctly in local directory.

(jLabs) tklicki@tklicki-dev:~/jLabs$ pip install jsnapy
Collecting jsnapy
  Downloading jsnapy-1.0.0.tar.gz
Requirement already satisfied: junos-eznc in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from jsnapy)
Collecting colorama (from jsnapy)
  Using cached colorama-0.3.7-py2.py3-none-any.whl
Collecting ConfigParser (from jsnapy)
  Downloading configparser-3.5.0.tar.gz
Collecting pyparsing (from jsnapy)
  Downloading pyparsing-2.1.10-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 4.7MB/s
Collecting icdiff (from jsnapy)
  Downloading icdiff-1.8.5.tar.gz
Requirement already satisfied: netaddr in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: PyYAML>=3.10 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: pyserial in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: scp>=0.7.0 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: jinja2>=2.7.1 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: six in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: paramiko>=1.15.2 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: lxml>=3.2.4 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: ncclient>=0.4.6 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from junos-eznc->jsnapy)
Requirement already satisfied: MarkupSafe in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from jinja2>=2.7.1->junos-eznc->jsnapy)
Requirement already satisfied: pyasn1>=0.1.7 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from paramiko>=1.15.2->junos-eznc->jsnapy)
Requirement already satisfied: cryptography>=1.1 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from paramiko>=1.15.2->junos-eznc->jsnapy)
Requirement already satisfied: setuptools>0.6 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from ncclient>=0.4.6->junos-eznc->jsnapy)
Requirement already satisfied: ipaddress in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc->jsnapy)
Requirement already satisfied: enum34 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc->jsnapy)
Requirement already satisfied: idna>=2.0 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc->jsnapy)
Requirement already satisfied: cffi>=1.4.1 in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc->jsnapy)
Requirement already satisfied: pycparser in /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages (from cffi>=1.4.1->cryptography>=1.1->paramiko>=1.15.2->junos-eznc->jsnapy)
Building wheels for collected packages: jsnapy, ConfigParser, icdiff
  Running setup.py bdist_wheel for jsnapy ... error
  Complete output from command /home/tklicki/.virtualenvs/jLabs/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-L2cIfS/jsnapy/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpHzAJsjpip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-2.7
  creating build/lib.linux-x86_64-2.7/jnpr
  copying lib/jnpr/__init__.py -> build/lib.linux-x86_64-2.7/jnpr
  creating build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/setup_logging.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/check.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/version.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/snap.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/xml_comparator.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/notify.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/sqlite_store.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/sqlite_get.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/jsnapy.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/__init__.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/testop.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/jsnapy.cfg -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/logging.yml -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  copying lib/jnpr/jsnapy/content.html -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
  running build_scripts
  creating build/scripts-2.7
  copying and adjusting tools/jsnap2py -> build/scripts-2.7
  changing mode of build/scripts-2.7/jsnap2py from 644 to 755
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/jnpr
  copying build/lib.linux-x86_64-2.7/jnpr/__init__.py -> build/bdist.linux-x86_64/wheel/jnpr
  creating build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/setup_logging.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/check.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/version.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/content.html -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/snap.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/xml_comparator.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/notify.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/sqlite_store.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/sqlite_get.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/logging.yml -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/jsnapy.cfg -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/jsnapy.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/__init__.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/testop.py -> build/bdist.linux-x86_64/wheel/jnpr/jsnapy
  running install_data
  creating build/bdist.linux-x86_64/wheel/jsnapy-1.0.0.data
  creating build/bdist.linux-x86_64/wheel/jsnapy-1.0.0.data/data
  creating build/bdist.linux-x86_64/wheel/etc
  creating build/bdist.linux-x86_64/wheel/etc/jsnapy
  copying lib/jnpr/jsnapy/logging.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy
  creating build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_is_equal.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_full_output.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/module_data.py -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/config_test_rpc.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_diff.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_wo_mssg.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_contains.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_not_more.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_all_same.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/config_multiple_snapcheck.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_xml.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/contains1.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/main.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_all_same2.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_filter.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/module_check.py -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_delta.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/module_snapcheck.py -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_not_in.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_not_exists.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/devices.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/send_mail.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_interface.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/config_check_no_test.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/config_multiple_check.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_no_diff.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/first_test.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/__init__.py -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_is_gt.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_is_in.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_not_less.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/config_single_snapcheck.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_in_range.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_not_equal.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_not_range.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_bgp_neighbor.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_is_lt.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/config_single_check.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/config_diff.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_exists.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/module_device.py -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying samples/test_get_config.yml -> build/bdist.linux-x86_64/wheel/etc/jsnapy/samples
  copying lib/jnpr/jsnapy/jsnapy.cfg -> build/bdist.linux-x86_64/wheel/etc/jsnapy
  creating build/bdist.linux-x86_64/wheel/etc/jsnapy/testfiles
  copying testfiles/README -> build/bdist.linux-x86_64/wheel/etc/jsnapy/testfiles
  creating build/bdist.linux-x86_64/wheel/etc/jsnapy/snapshots
  copying snapshots/README -> build/bdist.linux-x86_64/wheel/etc/jsnapy/snapshots
  creating build/bdist.linux-x86_64/wheel/var
  creating build/bdist.linux-x86_64/wheel/var/log
  creating build/bdist.linux-x86_64/wheel/var/log/jsnapy
  copying logs/debug.log -> build/bdist.linux-x86_64/wheel/var/log/jsnapy
  copying logs/jsnapy.log -> build/bdist.linux-x86_64/wheel/var/log/jsnapy
  copying logs/info.log -> build/bdist.linux-x86_64/wheel/var/log/jsnapy
  copying logs/errors.log -> build/bdist.linux-x86_64/wheel/var/log/jsnapy
  copying logs/__init__.py -> build/bdist.linux-x86_64/wheel/var/log/jsnapy
  copying logs/critical.log -> build/bdist.linux-x86_64/wheel/var/log/jsnapy
  running install_egg_info
  running egg_info
  writing requirements to lib/jsnapy.egg-info/requires.txt
  writing lib/jsnapy.egg-info/PKG-INFO
  writing top-level names to lib/jsnapy.egg-info/top_level.txt
  writing dependency_links to lib/jsnapy.egg-info/dependency_links.txt
  writing entry points to lib/jsnapy.egg-info/entry_points.txt
  warning: manifest_maker: standard file '-c' not found

  reading manifest file 'lib/jsnapy.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'lib/jsnapy.egg-info/SOURCES.txt'
  Copying lib/jsnapy.egg-info to build/bdist.linux-x86_64/wheel/jsnapy-1.0.0-py2.7.egg-info
  running install_scripts
  creating build/bdist.linux-x86_64/wheel/jsnapy-1.0.0.data/scripts
  copying build/scripts-2.7/jsnap2py -> build/bdist.linux-x86_64/wheel/jsnapy-1.0.0.data/scripts
  changing mode of build/bdist.linux-x86_64/wheel/jsnapy-1.0.0.data/scripts/jsnap2py to 755
  error: [Errno 2] No such file or directory: '/etc/jsnapy'

  ----------------------------------------
  Failed building wheel for jsnapy
  Running setup.py clean for jsnapy
  Running setup.py bdist_wheel for ConfigParser ... done
  Stored in directory: /home/tklicki/.cache/pip/wheels/1c/bd/b4/277af3f6c40645661b4cd1c21df26aca0f2e1e9714a1d4cda8
  Running setup.py bdist_wheel for icdiff ... done
  Stored in directory: /home/tklicki/.cache/pip/wheels/ff/e7/2c/3ecb5be52b75ddb49b5520a49da9c72124cf8d4c0ddb365ab0
Successfully built ConfigParser icdiff
Failed to build jsnapy
Installing collected packages: colorama, ConfigParser, pyparsing, icdiff, jsnapy
  Running setup.py install for jsnapy ... error
    Complete output from command /home/tklicki/.virtualenvs/jLabs/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-L2cIfS/jsnapy/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-V6nplV-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/tklicki/.virtualenvs/jLabs/include/site/python2.7/jsnapy:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-2.7
    creating build/lib.linux-x86_64-2.7/jnpr
    copying lib/jnpr/__init__.py -> build/lib.linux-x86_64-2.7/jnpr
    creating build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/setup_logging.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/check.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/version.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/snap.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/xml_comparator.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/notify.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/sqlite_store.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/sqlite_get.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/jsnapy.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/__init__.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/testop.py -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/jsnapy.cfg -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/logging.yml -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    copying lib/jnpr/jsnapy/content.html -> build/lib.linux-x86_64-2.7/jnpr/jsnapy
    running build_scripts
    creating build/scripts-2.7
    copying and adjusting tools/jsnap2py -> build/scripts-2.7
    changing mode of build/scripts-2.7/jsnap2py from 644 to 755
    running install_lib
    copying build/lib.linux-x86_64-2.7/jnpr/__init__.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr
    creating /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/setup_logging.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/check.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/version.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/content.html -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/snap.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/xml_comparator.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/notify.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/sqlite_store.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/sqlite_get.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/logging.yml -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/jsnapy.cfg -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/jsnapy.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/__init__.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    copying build/lib.linux-x86_64-2.7/jnpr/jsnapy/testop.py -> /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/__init__.py to __init__.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/setup_logging.py to setup_logging.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/check.py to check.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/version.py to version.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/snap.py to snap.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/xml_comparator.py to xml_comparator.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/notify.py to notify.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/sqlite_store.py to sqlite_store.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/sqlite_get.py to sqlite_get.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/jsnapy.py to jsnapy.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/__init__.py to __init__.pyc
    byte-compiling /home/tklicki/.virtualenvs/jLabs/lib/python2.7/site-packages/jnpr/jsnapy/testop.py to testop.pyc
    running install_data
    creating /etc/jsnapy
    error: could not create '/etc/jsnapy': Permission denied

    ----------------------------------------
Command "/home/tklicki/.virtualenvs/jLabs/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-L2cIfS/jsnapy/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-V6nplV-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/tklicki/.virtualenvs/jLabs/include/site/python2.7/jsnapy" failed with error code 1 in /tmp/pip-build-L2cIfS/jsnapy/

**Comment by eeishaan** [comment]
A pull request was merged regarding this: #179 
Please refer #179 and check if the installation method works for you or not. Though #179 doesn't directly addresses your problem, in the meantime you can do the following:
1. Download the latest code and install it as given in #179 to a directory of your liking.
2. After installing, copy the `jsnapy.cfg` and `logging.yml` from the source code and place it in `~/.jsnapy`. You can choose any other directory as well, just make sure to set an environment variable `JSNAPY_HOME` that points to it.

**Comment by XioNoX** [comment]
I'm having the same issue, but can't install it like indicated in #179. Both
$ python setup.py install --install-data /home/xionox/network-automation/ (after a git clone of current master)
and
$ pip install jsnapy --install-option="--install-data=/home/xionox/network-automation/
end with:
```
creating /etc/jsnapy
error: could not create '/etc/jsnapy': Permission denied
```
I also tried to set the ENV variable
```
echo $JSNAPY_HOME
/home/xionox/network-automation
```
Any ideas?

**Comment by eeishaan** [comment]
Hi @XioNoX 
The problem is that jsnapy tries to create /etc/jsnapy directory and places jsnapy.cfg and logging.yml file inside it.
This is default behavior and these two files are not affected by the above methods.

I'll modify the code so that if JSNAPY_HOME is set before installation, it picks up that location as default and places jsnapy.cfg and logging.yml inside it.

In the meantime you can try pip with sudo. I'll issue a fix by tomorrow for this.

**Comment by XioNoX** [comment]
@eeishaan Thanks! Any idea when this will be released?

**Comment by ruissalo** [comment]
This is actually a very annoying issue that breaks current python packaging best practices and makes it impossible to use jsnapy as a regular library, in particular, if you are using any sort of on-demand environment. There shouldn't be any dependencies to external configuration files.

**Comment by eeishaan** [comment]
@ruissalo we have made the changes and they are under review as #242 . They will merged into the master soon. Things will be much smoother and in line with the best practices after this.

**Comment by vnitinv** [comment]
Fixed via #246

**Comment by hopisaurus** [comment]
Any chance we can get this pushed out to PyPi as a release? Really defeats the purpose of the virtual environments if we cannot set them install directories to custom...
